mono_repo/lib/src/commands/github/github_yaml.dart

```diff
@@ -275,13 +277,25 @@ extension on CIJobEntry {
         workingDirectory: package,
       ));
       for (var i = 0; i < commands.length; i++) {
+        final task = job.tasks[i];
+        var workingDirectory = task.action?.workingDirectory;
+        if (workingDirectory != null) {
+          workingDirectory = posix.normalize(
+            posix.join(package, workingDirectory),
+          );
+        }
         commandEntries.add(_CommandEntry(
-          '$package; ${job.tasks[i].command}',
-          _commandForOs(job.tasks[i].command),
-          // Run this regardless of the success of other steps other than the
-          // pub step.
-          ifCondition: "always() && steps.$pubStepId.conclusion == 'success'",
-          workingDirectory: package,
+          '$package; ${task.command}',
+          _commandForOs(task.command),
+          id: task.action?.id,
```

Comment:
That's a good question. At that point, does it make sense for `action` to function more similar to `command` and spread its JSON into the GitHub action yaml so that, for example, overriding `run` becomes possible? Right now, the following is unclear in my mind:

```yaml
stages:
  - custom_step:
      - command: ./script_a
        action:
            run: ./script_b
```

But this could possibly make more sense:

```yaml
stages:
  - custom_step:
      - action:
           run: ./script_b
```

Reply:
Yeah, possibly just allowing a fully custom action task would make the most sense? I kind of like that.

Reply:
Sweet! I’ll work on updating it 👍

Reply:
Okay, so I've made the following changes:
- `action` now functions similar to `command` as a top level step type
- Consolidated GitHub action config into the `GitHubActionConfig` class since there was a lot of overlap with `_CommandEntry`
- Added some validators for known GitHub step config keys (`run`, `uses`, `with`, etc.) but allows for unknown keys to be used as well. Let me know if these helpers are useful or if we should just dump whatever we're given. There is the possibility for things to change, and the flipside is that I think it offers a bit nicer DX than discovering a typo down the road.